ObjectivePGP/CryptoBox/PGPEC.m

```diff
@@ -255,8 +255,10 @@ + (BOOL)publicEncrypt:(nonnull NSData *)data withPublicKeyPacket:(PGPPublicKeyPa
                 return @[];
             }
 
+            NSData* hash = [toSign pgp_SHA512];
```

Comment:
Why exactly this is SHA512? It seems like this is a key/signature parameter than hardcoded (I failed to find relevant rfc section)

Reply:
You can right, but in this rfc: https://www.rfc-editor.org/rfc/rfc8032 SHA-512 is mentioned. It can be refined to handle more hashed based on hashAlgorithm easily, since e.g. BouncyCastle in Java handle this way also.